api/v1/checks.go

```diff
@@ -125,17 +136,13 @@ type Bucket struct {
 }
 
 type S3Check struct {
-	Description `yaml:",inline" json:",inline"`
-	Bucket      Bucket `yaml:"bucket" json:"bucket,omitempty"`
-	AccessKey   string `yaml:"accessKey" json:"accessKey,omitempty"`
-	SecretKey   string `yaml:"secretKey" json:"secretKey,omitempty"`
-	ObjectPath  string `yaml:"objectPath" json:"objectPath,omitempty"`
-	// Skip TLS verify when connecting to s3
-	SkipTLSVerify bool `yaml:"skipTLSVerify,omitempty" json:"skipTLSVerify,omitempty"`
+	Description   `yaml:",inline" json:",inline"`
+	AWSConnection `yaml:",inline" json:",inline"`
+	BucketName    string `yaml:"bucketName" json:"bucketName,omitempty"`
```

Comment:
Why the rename of the field ?

Reply:
Because the structure itself has changed. Before "bucket" was a struct

```yaml
  s3:
    - bucket:
        name: "test-bucket"
        region: "us-east-1"
        endpoint: "https://test-bucket.s3.us-east-1.amazonaws.com"
      secretKey: "****************"
      accessKey: "~~~~~~~~~~~~~~~~"
      objectPath: "path/to/object"
      skipTLSVerify: true
```

but I changed it so AWSConnection can be used like we do for all other AWS related checks.

I can rename to `bucket` but a change to YAML files are needed regardless.